### PR TITLE
Add affordability calculator

### PR DESCRIPTION
**Pull request by chris-pardy**
This computes the maximum loan size affordable to hit a target
monthly payment. Additional monthly payments can be modeled as
a series of discreet linear cost models.

### CODE REVIEW

File: src/affordability.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+/**
+ * Inputs to the computeAffordability function
+ */
+export type ComputeAffordabilityInputs<TCostLabels extends string> = Omit<
+  ComputeMonthlyPaymentInputs,
+  'presentValue'
+> & {
+  /**
+   * Cost models for monthly costs costs.
```
Comment (kisaraofpern) [inline src/affordability.ts:34]: Extra `costs`.

File: src/affordability.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    [TLabel in TCostLabels | 'pAndI']: number;
+  };
+};
+
+type ComputableModel<TCostLabels extends string> = CostModel & {
+  compute: (costs: Record<TCostLabels, number>, presentValue: number) => void;
+};
+
+// project the compute function on the cost model.
+const convertToComputableModels = <TCostLabels extends string>(
```
Comment (kisaraofpern) [inline src/affordability.ts:65]: I think `TCostLabel` should be singular here?

File: src/affordability.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+   */
+  presentValueFactor: number;
+  /**
+   * A fixed offset of monthly cost to add to the result of the presentValueFactor * presentValue
+   */
+  costOffset: number;
+  /**
+   * A minimum present value for this cost model to be valid
+   */
+  minPresentValue: number;
```
Comment (kisaraofpern) [inline src/affordability.ts]: To make sure I'm understanding this correctly, in the case of PMI, this value would be the loan amount equivalent to the value defined by the lower bound of the percentage range?

E.g., for loan amount $500_000, PMI rate of 1% for 3-5%, this value would be $15_000?

Comment (chris-pardy) [reply src/affordability.ts]: Sort of it's easier to explain with something like taxes:
* Imagine you had a tax rate of 1% for Homes under $1,000,000 and a rate of 1.2% for homes over 1,000,000
* Imagine you had a $200,000 downpayment
* You would create 2 cost model elements, one with a `maxPresentValue` of $800,000 and the other with a `maxPresentValue` of Positive Infinity. The rates would be 0.01 and 0.012 respectively. The cost offset would be $2,000 and $2,400 respectively.

**Review by kisaraofpern — APPROVED** [review]
Looks good, as far as I can tell! :)
Looking forward to putting it through its paces.

**Pull request merged**